projects/storefrontlib/src/cms-components/product/product-list/product-facet-navigation/facet/facet.component.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+} from '@angular/core';
+import { Facet, FacetValue } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { ICON_TYPE } from '../../../../../cms-components/misc/icon/icon.model';
+import { FacetCollapseState } from '../facet.model';
+import { FacetService } from '../services/facet.service';
+
+@Component({
+  selector: 'cx-facet',
+  templateUrl: './facet.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FacetComponent {
+  protected _facet: Facet;
+
+  state$: Observable<FacetCollapseState>;
+
+  expandedIcon = ICON_TYPE.EXPAND;
+
+  /** configurable icon that is used to collapse the facet group  */
+  @Input() expandIcon: ICON_TYPE = ICON_TYPE.EXPAND;
+  @Input() collapseIcon: ICON_TYPE = ICON_TYPE.COLLAPSE;
+
+  @HostBinding('class.expanded') isExpanded: boolean;
+  @HostBinding('class.multi-select') isMultiSelect: boolean;
+
+  @Input()
+  set facet(value: Facet) {
+    this._facet = value;
+    this.isMultiSelect = !!value.multiSelect;
+
+    const state$ = this.facetService.getState(value);
+    // the initial observed state doesn't work without this...
+    this.isExpanded = state$.value.expanded ?? state$.value.expandBydefault;
+    this.state$ = state$.pipe(
+      tap(state => (this.isExpanded = state.expanded ?? state.expandBydefault))
+    );
+  }
+  get facet(): Facet {
+    return this._facet;
+  }
+
+  @Input() @HostBinding('attr.class') class: string;
+
+  constructor(
+    protected facetService: FacetService,
+    protected elementRef: ElementRef<HTMLElement>
+  ) {}
+
+  /**
+   * Handles clicking the heading of the facet group, which means toggling the visibility
+   * of the group (collapse / expand) and optionally focusing the group.
+   */
+  toggleGroup(event: UIEvent) {
+    const host: HTMLElement = this.elementRef.nativeElement;
+    const isLocked = host.classList.contains('is-locked');
```

Comment:
P2: I guess we depend on the implementation detail of the `cxFocus` directive - that it sets the `is-locked` class on the common host element. Please correct me if I'm wrong. A small code comment with explanantion will be useful, anyway.

Did you consider using dependency injection to inject here `@Optional` focus directive and use it's public API to learn the i.e. `isLocked` state. Then we won't depend on the implementation detail of the cxFocus (which is a CSS class - that is hard to reason about, since anyone can set css classes), but explicitly the public API of focus directive class.

Reply:
thanks for bringing this idea to the table. I've followed up on your suggestion, but use viewChild instead:

`@ViewChild(FocusDirective) keyboardFocus: FocusDirective;`

I like that more as we do not have the additional constructor argument.